nixos/modules/services/matrix/mautrix-whatsapp.nix

```
@@ -49,74 +70,87 @@ let
-          };
-          provisioning = {
-            shared_secret = "disable";
-          };
+          mute_only_on_create = false;
           permissions = {
             "example.com" = "user";
           };
         };
+        database = {
+          type = "postgres";
+          uri = "postgresql:///mautrix_whatsapp?host=/run/postgresql";
+        };
+        homeserver = {
+          address = "http://[::1]:8008";
+          domain = "my-domain.tld";
+        };
+        appservice = {
+          id = "whatsapp";
+          ephemeral_events = false;
+        };
+        matrix.message_status_events = true;
+        provisioning = {
+          shared_secret = "disable";
+        };
+        backfill.enabled = true;
+        encryption = {
+          allow = true;
+          default = true;
+          require = true;
+          pickle_key = "$ENCRYPTION_PICKLE_KEY";
+        };
       };
     };
+
     environmentFile = lib.mkOption {
       type = lib.types.nullOr lib.types.path;
       default = null;
       description = ''
-        File containing environment variables to be passed to the mautrix-whatsapp service,
-        in which secret tokens can be specified securely by optionally defining a value for
-        `MAUTRIX_WHATSAPP_BRIDGE_LOGIN_SHARED_SECRET`.
+        File containing environment variables to be passed to the mautrix-whatsapp service.
+        If an environment variable `MAUTRIX_WHATSAPP_BRIDGE_LOGIN_SHARED_SECRET` is set,
+        then its value will be used in the configuration file for the option
+        `double_puppet.secrets` without leaking it to the store, using the configured
+        `homeserver.domain` as key.
       '';
     };
 
     serviceDependencies = lib.mkOption {
       type = with lib.types; listOf str;
-      default = lib.optional config.services.matrix-synapse.enable config.services.matrix-synapse.serviceUnit;
+      default =
+        (lib.optional config.services.matrix-synapse.enable config.services.matrix-synapse.serviceUnit)
+        ++ (lib.optional config.services.matrix-conduit.enable "conduit.service");
       defaultText = lib.literalExpression ''
-        optional config.services.matrix-synapse.enable config.services.matrix-synapse.serviceUnits
+        (optional config.services.matrix-synapse.enable config.services.matrix-synapse.serviceUnit)
+        ++ (optional config.services.matrix-conduit.enable "conduit.service")
       '';
       description = ''
-        List of Systemd services to require and wait for when starting the application service.
+        List of systemd units to require and wait for when starting the application service.
       '';
     };
 
     registerToSynapse = lib.mkOption {
       type = lib.types.bool;
       default = config.services.matrix-synapse.enable;
-      defaultText = lib.literalExpression "config.services.matrix-synapse.enable";
+      defaultText = lib.literalExpression ''
+        config.services.matrix-synapse.enable
+      '';
```

```suggestion
      defaultText = lib.literalExpression "config.services.matrix-synapse.enable";
```
Otherwise we get extra new lines which are technically wrong